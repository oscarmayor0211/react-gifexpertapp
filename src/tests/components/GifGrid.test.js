import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import React from 'react';
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");
describe('Prueba con el <GifGrid />', () => {

    const category = 'One Punch'
    test('debe de mostrar  <GifGrid /> correctamente', () => {
        
        useFetchGifs.mockReturnValue({

            data: [],
            loading : true
        })
        const wrapper = shallow(<GifGrid category={category} />);
        expect( wrapper).toMatchSnapshot();
    }),

    test('Debe de mostrar items cuando se cargan imagenes al useFetchGifs ', () => {

        const gifs = [{
            id : 'ABC',
            url :'https://localhost/cualquier/cosa.jpg',
            title : 'cualquier cosa'
        }]
        useFetchGifs.mockReturnValue({

            data: gifs,
            loading : false
        })
        const wrapper = shallow(<GifGrid category={category} />);
        expect( wrapper).toMatchSnapshot();
        expect( wrapper.find('p').exists()).toBe(false);
        expect( wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
    
})
