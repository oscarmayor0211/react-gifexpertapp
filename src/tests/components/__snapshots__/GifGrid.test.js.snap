// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba con el <GifGrid /> Debe de mostrar items cuando se cargan imagenes al useFetchGifs  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Prueba con el <GifGrid /> debe de mostrar  <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading ..
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
